modules/solid_mechanics/src/materials/IsotropicTempDepHardening.C

```diff
@@ -0,0 +1,142 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "IsotropicTempDepHardening.h"
+
+#include "PiecewiseLinear.h"
+
+template<>
+InputParameters validParams<IsotropicTempDepHardening>()
+{
+  InputParameters params = validParams<IsotropicPlasticity>();
+
+  params.set<Real>("yield_stress") = 1.0;
+  params.set<Real>("hardening_constant") = 1.0;
+
+  params.suppressParameter<Real>("yield_stress");
+  params.suppressParameter<FunctionName>("yield_stress_function");
+  params.suppressParameter<Real>("hardening_constant");
+  params.suppressParameter<FunctionName>("hardening_function");
+
+  params.addRequiredParam<std::vector<FunctionName> >("hardening_functions", "List of functions of true stress as function of plastic strain at different temperatures");
+  params.addRequiredParam<std::vector<Real> >("temperatures", "List of temperatures for the true stress as a function of plastic strain curves");
+
```

Comment:
I think I'd change the description to something like:
"List of temperatures corresponding to the functions listed in 'hardening_functions'"